internal/indexer/indexer.go

```diff
@@ -168,7 +168,7 @@ func RunningPromotionsByArgoCDApplications(
 				Promotion: promo.Name,
 				Vars:      promo.Spec.Vars,
 			}
-			vars, err := dirStep.GetVars(promoCtx)
+			vars, err := dirStep.GetVars(promoCtx, promoCtx.State)
```

Comment:
Since we only built out enough of promoCtx to get us through this, promotCtx.State will be empty. I suppose that's actually ok in this (pardon the overload) _context?_

Reply:
Explicit nil sgtm, maybe with a brief comment about why that arg is unimportant in this context?

Really this is a nit... so however you want to deal with it is fine, even if it's do nothing.

Reply:
Upon closer inspection, I do think we need to pass some state here. In addition, I have also a vague suspicion we may actually need the outputs as well while evaluating the individual fields because _theoretically_, the values can not just originate from variables.